Scripts/RemediationScripts/Remediate-EnableSecurityScanningForContainerRegistry.ps1

```diff
@@ -180,116 +193,192 @@ function Enable-SecurityScanningIdentityForContainerRegistry
-            $resourceId = $_.ResourceId
-
+        Write-Host "Fetching all Container Registry(s) failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No Container Registry(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
             try
             {
                 $containerRegistryResource =  Get-AzResource -ResourceType Microsoft.ContainerRegistry/registries -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
-                
                 $containerRegistryDetails += $containerRegistryResource  | Select-Object @{N='ResourceId';E={$_.ResourceId}},
-                                                                          @{N='ResourceGroupName';E={$_.ResourceGroupName}},
-                                                                          @{N='ResourceName';E={$_.Name}},
-                                                                          @{N='ObjectId';E={$ObjectId}}
+                                                                            @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                            @{N='ResourceName';E={$_.ResourceName}},
+                                                                            @{N='ObjectId';E={$ObjectId}}
             }
             catch
             {
-                Write-Host "Error fetching Container Registry(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping the Resource: [$($_.ResourceName)]..." -ForegroundColor $([Constants]::MessageType.Warning)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    
+                $logSkippedResources += $logResource
+                Write-Host $([Constants]::SingleDashLine)
```

Comment:
Doble dash line

Reply:
not needed here as it is not exiting the code at this point